com.microsoft.mrtk.windowsspeech/Subsystems/WindowsSpeechRecognitionSubsystem.cs

```diff
@@ -0,0 +1,256 @@
+        [Preserve]
+        class WindowsSpeechRecognitionProvider : Provider
+        {
+#if UNITY_EDITOR_WIN || UNITY_STANDALONE_WIN || UNITY_WSA
+            /// <summary>
+            /// The confidence threshold for the recognizer to return its result.
+            /// </summary>
+            public ConfidenceLevel ConfidenceLevel => confidenceLevel;
+
+            /// <summary>
+            /// The time length in seconds before dictation recognizer session ends due to lack of audio input in case there was no audio heard in the current session.
+            /// </summary>
+            public float InitialSilenceTimeoutSeconds
+            {
+                get => initialSilenceTimeoutSeconds;
+                set
+                {
+                    if (initialSilenceTimeoutSeconds != value)
+                    {
+                        initialSilenceTimeoutSeconds = value;
+                        if (dictationRecognizer != null)
+                        {
+                            dictationRecognizer.InitialSilenceTimeoutSeconds = value;
+                        }
+                    }
+                }
+            }
+
+            /// <summary>
+            /// The time length in seconds before dictation recognizer session ends due to lack of audio input.
+            /// </summary>
+            public float AutoSilenceTimeout
+            {
+                get => autoSilenceTimeout;
+                set
+                {
+                    if (autoSilenceTimeout != value)
+                    {
+                        autoSilenceTimeout = value;
+                        if (dictationRecognizer != null)
+                        {
+                            dictationRecognizer.AutoSilenceTimeoutSeconds = value;
+                        }
+                    }
+                }
+            }
+
+            private WindowsSpeechRecognitionSubsystemConfig config;
+            private DictationRecognizer dictationRecognizer;
+            private ConfidenceLevel confidenceLevel;
+            private float initialSilenceTimeoutSeconds;
+            private float autoSilenceTimeout;
+#endif
+
+            /// <summary>
+            /// Constructor of WindowsSpeechRecognitionProvider.
+            /// </summary>
+            public WindowsSpeechRecognitionProvider()
+            {
+#if !(UNITY_EDITOR_WIN || UNITY_STANDALONE_WIN || UNITY_WSA)
+                Debug.LogError("Cannot create WindowsSpeechRecognitionSubsystem because WindowsSpeechRecognitionSubsystem is only supported on Windows Editor, Standalone Windows and UWP.");
+#endif
+            }
+
+#if UNITY_EDITOR_WIN || UNITY_STANDALONE_WIN || UNITY_WSA
+            /// <inheritdoc/>
+            public override void Start()
+            {
+                base.Start();
+                config = XRSubsystemHelpers.GetConfiguration<WindowsSpeechRecognitionSubsystemConfig, WindowsSpeechRecognitionSubsystem>();
+                confidenceLevel = config.ConfidenceLevel;
```

Comment:
If a subsystem happens to be stopped and restarted at runtime, do we want the local state to be overwritten? Or do we want the previous runtime customization to persist?

Reply:
Great question... I wonder if there is a way for us to know whether the Start is being called by the initialization code or some other code sometime later? I guess we can use a bool to track (not elegant unfortunately). But I think maybe it's fair to say when a subsystem restarts it reads from its config file. Also this is something we should standardize across all subsystems as it's not specific to this one.

Reply:
what about applying configuration data in the ctor and not in start? that way, as long as the subsystem is not torn down, any config changes would remain.

Reply:
I had a chat with @Zee2 and we are leaning towards making the subsystem reads its config every time it starts, regardless of whether it's the first start or subsequent ones.

Reply:
is this something that should retroactively be added to _all_ subsystems?

Reply:
We should make our subsystems have consistent behavior on this aspect.